client/src/components/Viewer/Viewer.js

```diff
@@ -55,6 +55,9 @@ function Viewer(props) {
         setIsConnected(true);
         socket.emit('addUser', {
           token: JSON.parse(localStorage.getItem('dfs-user'))?.['token'],
+          user_id: JSON.parse(localStorage.getItem('dfs-user'))?.['user'][
+            'user_email'
+          ],
```

Comment:
The addition of the `user_id` field is a good enhancement, but ensure that the `user_email` field exists in the stored user information to avoid potential runtime errors. Consider adding a check to handle cases where the user data might not be available or is structured differently. This will improve the robustness of the code.


    *This comment was auto-generated by **Code Bot** to provide suggestions and improvements.*